### ISSUE TITLE
Rule request: avoid boilerplate string litterals.

### ISSUE THREAD

**Issue by njourdane** [astral-sh/ruff#21144]
### Summary

Instead of:

```py
if c in "abcdefghijklmnopqrstuvwxyz":
    print(c)
```

Use:

```py
from string import ascii_lowercase
if c in ascii_lowercase:
    print(c)
```

And more generally:

- `abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ` -> `string.ascii_letters`
- `abcdefghijklmnopqrstuvwxyz` -> `string.ascii_lowercase`
- `ABCDEFGHIJKLMNOPQRSTUVWXYZ` -> `string.ascii_uppercase`
- `0123456789` -> `string.digits`
- `0123456789abcdefABCDEF` -> `string.hexdigits`
- `01234567` -> `string.octdigits`

The main reason is, when using a string literal like `abcdefghijklmopqrstuvwxyz` it's not abvious when a letter is missing (oops, just did it, did you see it?), so it's a potential source of errors.

**Comment by ntBre** [comment]
This makes sense to me. Clippy has a similar [`manual_is_ascii_check`](https://rust-lang.github.io/rust-clippy/master/index.html#manual_is_ascii_check) too.

**Comment by ntBre** [comment]
Oh, actually we have an existing rule for this in preview: [hardcoded-string-charset (FURB156)](https://docs.astral.sh/ruff/rules/hardcoded-string-charset/#hardcoded-string-charset-furb156).

**Comment by lypwig** [comment]
Great!

A suggestion: when using the `in` keyword, the rule could work whatever the letters order.

Ie. these lines are similar*, but only the first one is detected by the linter:

<img width="235" height="64" alt="Image" src="https://github.com/user-attachments/assets/918002ce-6c6e-41c0-94de-90655e8688e4" />

https://play.ruff.rs/36ad710a-cafd-42ba-8dce-6778299bdbe5

*It's not exactly the same because the iteration order differs, but I guess it's a very specific use-case to check in a specific order.

**Comment by ntBre** [comment]
Ah that's a good point. There's code in the rule for constructing bitsets of characters, but then we don't convert the input to a set for the comparison. The set field is actually completely unused from what I can tell.

**Comment by ntBre** [comment]
I looked into this a bit more and was going to make the set change, but I found https://github.com/astral-sh/ruff/issues/13802 and #14233. It's actually intentional that we only perform the check for that exact order because otherwise the diagnostic and fix is likely to be incorrect if `x` in your example isn't a single character:

```pycon
>>> import string
>>> '1' in '1234567890'
True
>>> '012' in '1234567890'
False
>>> '012' in string.digits
True
```

So I'll close this for now.
